File: app/src/main/java/com/twiceyuan/galleryviewpager/MainActivity.java
package com.twiceyuan.galleryviewpager;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.twiceyuan.galleryviewpager.infiniteViewPager.InfinitePagerAdapter;
import com.twiceyuan.galleryviewpager.infiniteViewPager.InfiniteViewPager;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        InfiniteViewPager viewPager = (InfiniteViewPager) findViewById(R.id.id_viewpager);
        viewPager.setOffscreenPageLimit(3);
        viewPager.setAdapter(new InfinitePagerAdapter(new ImagePagerAdapter()));
        viewPager.setPageTransformer(true, new GalleryTransformer());
    }
}

File: app/src/main/java/com/twiceyuan/galleryviewpager/MainActivity.kt
package com.twiceyuan.galleryviewpager

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.twiceyuan.galleryviewpager.infiniteViewPager.InfinitePagerAdapter
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        viewPager.offscreenPageLimit = 3
        viewPager.adapter = InfinitePagerAdapter(ImagePagerAdapter())
        viewPager.setPageTransformer(true, GalleryTransformer)
    }
}
